lib/bolt/transport/sudoable/file_commands.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Bolt
+  module Transport
+    class Sudoable < Base
+      class FileCommands
+        def initialize(node, path)
+          @conn = node
+          @owner = node.user
+          @path = path
+          @logger = node.logger
+        end
+
+        def to_s
+          @path
+        end
+
+        def mkdirs(subdirs)
+          abs_subdirs = subdirs.map { |subdir| File.join(@path, subdir) }
+          result = @conn.execute(['mkdir', '-p'] + abs_subdirs)
+          if result.exit_code != 0
+            message = "Could not create subdirectories in '#{@path}': #{result.stderr.string}"
+            raise Bolt::Node::FileError.new(message, 'MKDIR_ERROR')
+          end
+        end
+
+        def chown(owner)
+          # Windows doesn't support local sudo
+          return if owner.nil? || owner == @owner || Bolt::Util.windows?
+
+          result = @conn.execute(['id', '-g', owner])
+          if result.exit_code != 0
+            message = "Could not identify group of user #{owner}: #{result.stderr.string}"
+            raise Bolt::Node::FileError.new(message, 'ID_ERROR')
+          end
+          group = result.stdout.string.chomp
+
+          # Chown can only be run by root.
+          result = @conn.execute(['chown', '-R', "#{owner}:#{group}", @path], sudoable: true, run_as: 'root')
+          if result.exit_code != 0
+            message = "Could not change owner of '#{@path}' to #{owner}: #{result.stderr.string}"
+            raise Bolt::Node::FileError.new(message, 'CHOWN_ERROR')
+          end
+
+          # File ownership successfully changed, record the new owner.
+          @owner = owner
+        end
+
+        def copy_file(destination)
```

Comment:
I think this should be removed/refactored? It is kind of confusing having this be used by `Bolt::Transport::Local::Shell#copy_file`. It does not work for macos 
```
u1u186j1hflhufg:bolt root# bundle exec bolt task run facts -n localhost
Started on localhost...
Failed on localhost:
  Could not copy file to /tmp/cc0a2936-9081-408b-90d5-7deee60410fc/bash.sh: cp: illegal option -- -
  usage: cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file target_file
         cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file ... target_directory
Failed on 1 node: localhost
Ran on 1 node in 0.12 seconds
u1u186j1hflhufg:bolt root# uname -a
Darwin u1u186j1hflhufg 17.3.0 Darwin Kernel Version 17.3.0: Thu Nov  9 18:09:22 PST 2017; root:xnu-4570.31.3~1/RELEASE_X86_64 x86_64
```

Also is there a different approach for handling the `write_remote_file` and `copy_file` using `alias`? It seems kind of awkward (but maybe its just a common ruby thing that i'm just not used to seeing).